From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Thu, 25 Aug 2011 16:51:28 -0700
Patch-mainline: v3.2-rc1
Subject: [PATCH] Staging: hv: hv_mouse: fix up remove() function
Git-commit: 6a180978aea7cb231c60b69b078ec339f40e2faa

This function obviously never worked, it was a cut-and-paste from the
probe() function.  Rewrite it to do what it is supposed to do at least
in theory.  As to if it works yet, well, it's not worse than it was
before, so this can't hurt.

Cc: K. Y. Srinivasan <kys@microsoft.com>
Cc: Haiyang Zhang <haiyangz@microsoft.com>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/hv_mouse.c |   10 ++--------
 1 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/hv/hv_mouse.c b/drivers/staging/hv/hv_mouse.c
index 083f28f..090736a 100644
--- a/drivers/staging/hv/hv_mouse.c
+++ b/drivers/staging/hv/hv_mouse.c
@@ -847,15 +847,10 @@ static int mousevsc_probe(struct hv_device *dev)
 
 static int mousevsc_remove(struct hv_device *dev)
 {
-	int ret = 0;
-
 	struct input_device_context *input_dev_ctx;
+	int ret;
 
-	input_dev_ctx = kmalloc(sizeof(struct input_device_context),
-				GFP_KERNEL);
-
-	dev_set_drvdata(&dev->device, input_dev_ctx);
-
+	input_dev_ctx = dev_get_drvdata(&dev->device);
 	if (input_dev_ctx->connected) {
 		hidinput_disconnect(input_dev_ctx->hid_device);
 		input_dev_ctx->connected = 0;
@@ -866,7 +861,6 @@ static int mousevsc_remove(struct hv_device *dev)
 	 * is being removed
 	 */
 	ret = mousevsc_on_device_remove(dev);
-
 	if (ret != 0) {
 		DPRINT_ERR(INPUTVSC_DRV,
 			   "unable to remove vsc device (ret %d)", ret);
-- 
1.6.0.2


